xarray/core/datatree.py

```diff
@@ -1064,14 +1059,18 @@ def from_dict(
 
         # First create the root node
         root_data = d.pop("/", None)
-        obj = cls(name=name, data=root_data, parent=None, children=None)
+        if isinstance(root_data, DataTree):
```

Comment:
The extra code here is a duplication of what is happening for the child nodes. `mypy` was unhappy with the types for the `root_data`, in case it was a `DataTree` instance.